src/main/java/hudson/remoting/Engine.java

```diff
@@ -203,14 +206,27 @@ public Thread newThread(final Runnable r) {
 
     private DelegatingX509ExtendedTrustManager agentTrustManager = new DelegatingX509ExtendedTrustManager(new BlindTrustX509ExtendedTrustManager());
 
+    private String directUrl;
```

Comment:
Name it differently, since it is not a "URL" (only host:port)?
The CLI parameter is called `-direct`.